File: Fibonacci/Java/Fibonacci_v4.java
package fibonacci_v4;

import java.util.Scanner;

public class Fibonacci_v4 {

    public static long fibonacci(int n) {
        if (n <= 2) {
            return n - 1;
        }
        return fibonacci(n - 1) + fibonacci(n - 2);
    }

    public static void main(String[] args) {
        int n;
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese número de términos: ");
        n = reader.nextInt();

        if (n <= 0) {
            System.out.printf("Debe ingresar un número mayor que cero");
        } else {
            int i;
            for (i = 1; i <= n; i++) {
                System.out.printf("%d ", fibonacci(i));
            }
        }
    }
}

File: Fibonacci/Python/fibonacci_v4.py
import numpy

def inicializar_arreglo(fibonacci, n):
    fibonacci[1] = 0;
    fibonacci[2] = 1;
    for i in range(3, n+1):
        fibonacci[i] = fibonacci[i - 1] + fibonacci[i - 2]
    
MAX_TERMINOS=100
n=int(input("Ingrese número de términos (1<=n<=100): "))
if n<=0:
    print("Debe ingresar un número en el rango [0..100]")
else:
    fibonacci=numpy.arange(MAX_TERMINOS+1)
    inicializar_arreglo(fibonacci, MAX_TERMINOS)
    for i in range(1, n+1):
        print("{:d} ".format(fibonacci[i]) , end="")